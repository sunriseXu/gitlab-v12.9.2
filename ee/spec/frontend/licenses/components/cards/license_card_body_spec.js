import { shallowMount } from '@vue/test-utils';
import LicenseCardBody from 'ee/licenses/components/cards/license_card_body.vue';

describe('LicenseCardBody', () => {
  let wrapper;
  const defaultProps = {
    license: {
      plan: 'ultimate',
      userLimit: 10,
      historicalMax: 20,
      overage: 5,
      startsAt: '2013/10/10',
      expiresAt: '2015/10/10',
      licensee: {
        Name: 'Jon Dough',
        Email: 'email@address.tanuki',
        Company: 'TanukiVille',
      },
    },
    isRemoving: false,
    activeUserCount: 10,
    guestUserCount: 8,
  };

  function createComponent(props = {}) {
    let propsData = props;
    propsData.license = Object.assign({}, defaultProps.license, props.license || {});
    propsData = Object.assign({}, defaultProps, props);

    wrapper = shallowMount(LicenseCardBody, {
      propsData,
    });
  }

  beforeEach(() => {
    jest.spyOn(global.Date.prototype, 'toString').mockReturnValue('2017/10/10');
  });

  afterEach(() => {
    if (wrapper) wrapper.destroy();
    global.Date.prototype.toString.mockRestore();
  });

  it('renders a license card body', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a license card body without free user info for non-ultimate licenses', () => {
    createComponent({ license: { plan: 'premium' } });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a loading state if isRemoving', () => {
    createComponent({ isRemoving: true });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders fallback licensee values', () => {
    createComponent({ licensee: {} });

    expect(wrapper.element).toMatchSnapshot();
  });
});
