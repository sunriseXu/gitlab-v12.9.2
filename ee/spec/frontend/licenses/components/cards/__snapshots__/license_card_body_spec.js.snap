// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseCardBody renders a license card body 1`] = `
<div
  class="card-body license-card-body p-0"
>
  <div
    class="license-table js-license-table"
  >
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="monitor"
        title="Usage"
      />
       
      <cell-stub
        isflexible="true"
        title="Seats in license"
        value="10"
      />
       
      <info-cell-stub
        popovercontent="Users with a Guest role or those who don't belong to any projects or groups don't count towards seats in use."
        title="Seats currently in use"
        value="2"
      />
       
      <info-cell-stub
        popovercontent="This is the maximum number of users that have existed at the same time since the license started. This is the minimum number of seats you will need to buy when you renew your license."
        title="Max seats used"
        value="20"
      />
       
      <info-cell-stub
        popovercontent="GitLab allows you to continue using your license even if you exceed the number of seats you purchased. You will be required to pay for these seats when you renew your license."
        title="Users outside of license"
        value="5"
      />
    </div>
     
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="calendar"
        title="Validity"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        title="Start date"
        value="2013/10/10"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        isexpirable="true"
        title="End date"
        value="2015/10/10"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        title="Uploaded on"
      />
    </div>
     
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="user"
        title="Registration"
      />
       
      <cell-stub
        isflexible="true"
        title="Licensed to"
        value="Jon Dough"
      />
       
      <cell-stub
        isflexible="true"
        title="Email address"
        value="email@address.tanuki"
      />
       
      <cell-stub
        isflexible="true"
        title="Company"
        value="TanukiVille"
      />
    </div>
  </div>
</div>
`;

exports[`LicenseCardBody renders a license card body without free user info for non-ultimate licenses 1`] = `
<div
  class="card-body license-card-body p-0"
>
  <div
    class="license-table js-license-table"
  >
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="monitor"
        title="Usage"
      />
       
      <cell-stub
        isflexible="true"
        title="Seats in license"
        value="10"
      />
       
      <cell-stub
        isflexible="true"
        popover-content="Users with a Guest role or those who don't belong to any projects or groups don't count towards seats in use."
        title="Seats currently in use"
        value="10"
      />
       
      <info-cell-stub
        popovercontent="This is the maximum number of users that have existed at the same time since the license started. This is the minimum number of seats you will need to buy when you renew your license."
        title="Max seats used"
        value="20"
      />
       
      <info-cell-stub
        popovercontent="GitLab allows you to continue using your license even if you exceed the number of seats you purchased. You will be required to pay for these seats when you renew your license."
        title="Users outside of license"
        value="5"
      />
    </div>
     
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="calendar"
        title="Validity"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        title="Start date"
        value="2013/10/10"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        isexpirable="true"
        title="End date"
        value="2015/10/10"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        title="Uploaded on"
      />
    </div>
     
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="user"
        title="Registration"
      />
       
      <cell-stub
        isflexible="true"
        title="Licensed to"
        value="Jon Dough"
      />
       
      <cell-stub
        isflexible="true"
        title="Email address"
        value="email@address.tanuki"
      />
       
      <cell-stub
        isflexible="true"
        title="Company"
        value="TanukiVille"
      />
    </div>
  </div>
</div>
`;

exports[`LicenseCardBody renders a loading state if isRemoving 1`] = `
<div
  class="card-body license-card-body p-0"
>
  <div
    class="p-5 d-flex justify-content-center align-items-center license-card-loading"
  >
    <icon-stub
      name="spinner"
      size="16"
    />
    <span
      class="ml-2"
    >
      Removing licenseâ€¦
    </span>
  </div>
</div>
`;

exports[`LicenseCardBody renders fallback licensee values 1`] = `
<div
  class="card-body license-card-body p-0"
  licensee="[object Object]"
>
  <div
    class="license-table js-license-table"
  >
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="monitor"
        title="Usage"
      />
       
      <cell-stub
        isflexible="true"
        title="Seats in license"
        value="10"
      />
       
      <info-cell-stub
        popovercontent="Users with a Guest role or those who don't belong to any projects or groups don't count towards seats in use."
        title="Seats currently in use"
        value="2"
      />
       
      <info-cell-stub
        popovercontent="This is the maximum number of users that have existed at the same time since the license started. This is the minimum number of seats you will need to buy when you renew your license."
        title="Max seats used"
        value="20"
      />
       
      <info-cell-stub
        popovercontent="GitLab allows you to continue using your license even if you exceed the number of seats you purchased. You will be required to pay for these seats when you renew your license."
        title="Users outside of license"
        value="5"
      />
    </div>
     
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="calendar"
        title="Validity"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        title="Start date"
        value="2013/10/10"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        isexpirable="true"
        title="End date"
        value="2015/10/10"
      />
       
      <date-cell-stub
        datenow="2017/10/10"
        title="Uploaded on"
      />
    </div>
     
    <div
      class="license-row d-flex"
    >
      <header-cell-stub
        icon="user"
        title="Registration"
      />
       
      <cell-stub
        isflexible="true"
        title="Licensed to"
        value="Jon Dough"
      />
       
      <cell-stub
        isflexible="true"
        title="Email address"
        value="email@address.tanuki"
      />
       
      <cell-stub
        isflexible="true"
        title="Company"
        value="TanukiVille"
      />
    </div>
  </div>
</div>
`;
